source/develop/release-management/features/virt/vm-lifecycle-with-kubevirt.md

```diff
@@ -0,0 +1,130 @@
+
+# The Current Process in oVirt
+
+In this section I elaborates on the process of running a virtual machine in oVirt. The creation of the virtual machine is done as a separate operation before it is being run so we can assume that the VM already exists in the database at this stage.  
+
+The process looks as follow:  
+0. Optional: a request is sent from client (UI/REST-API).  
+1. Instance of RunVmCommand is created.  
+2. The input is being verified. Specifically, the scheduler validates that there is a host capable of running the VM.  
+3. The engine retrieves all the properties of the VM to run.  
+4. The scheduler chooses a host to run the VM on.  
+5. The engine locks host devices, plug lun disks and update cinder disks.  
+6. The engine locks the monitoring thread.  
+7. The engine sends a request to VDSM on the chosen host to run the VM.  
+8. If the response if positive, the engine set the VM to WaitForLaunch state (and few other fields).  
+9. The monitoring lock is released.  
+10. The monitoring receives the VM as running (positive flow) or down (negative flow). If the VM fails to run, the engine tries to run it few more time (by default 2 additional attempts).
+
+# Proposed Approach
+
+So this proposal split the integration between oVirt and Kubernetes in regards to VM lifecycle into several phases. Note that some of them can be done in parallel.   
+
+## Milestone 1: Basic integration
+
+### Goal
+1. Add Kubevirt into oVirt.  
+2. Kubevirt sends the run-request to the relevant node.  
+
+### Design
+We aim to have Kubevirt as a configuration of a cluster so eventually we would be able to have a Data-center with cluster with lower compatibility version that works as it used to work before and cluster with the up-to-date compatibility version that works with Kubevirt. That way, we enable the user to upgrade existing system without having to stop the VMs.  
+
+So a cluster would have a property called "type". Initially the type will have two values: legacy, kubevirt. If 'kubvirt' is chosen, the user will need to provide the URL and authentication details of Kubevirt.  
+
+Note that we may use the same mechanism used in external provider to hold the authentication info. Another alternative is to use a full blown external provider for this, as we did for the integration with virt-v2v (where external provider is used only to contain pre-defined configuration of the remote system). Anyway, we do not have to decide this at this stage.  
+
+Then, the mentioned run-VM flow will be modified. The only change would be that in Kubvirt-cluster, CreateBrokerVDSCommand will generate a bit different configuration of the VM and will send it to Kubevirt (using the Kubernetes API) rather than VDSM. Note that the scheduling is still done in oVirt.  
```

Comment:
Should we note - eventually in a separate paragraph - that vdsm is still expected to run on every node in the cluster?